fabric-rendering-v1/src/main/java/net/fabricmc/fabric/api/client/rendering/v1/BuiltinItemRendererWithMode.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.rendering.v1;
+
+import net.minecraft.client.render.VertexConsumerProvider;
+import net.minecraft.client.render.model.json.ModelTransformation;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.item.ItemStack;
+
+/**
+ * Builtin item renderers render items with custom code.
+ * They allow using non-model rendering, such as BERs, for items.
+ *
+ * <p>An item with a builtin renderer must have a model extending {@code minecraft:builtin/entity}.
+ * The renderers are registered with {@link BuiltinItemRendererRegistry#register}.
+ */
+public interface BuiltinItemRendererWithMode {
+	/**
+	 * Renders an item stack.
+	 *
+	 * @param stack           the rendered item stack
+	 * @param mode            the model transformation mode
+	 * @param matrices        the matrix stack
+	 * @param vertexConsumers the vertex consumer provider
+	 * @param light           the color light multiplier at the rendering position
```

Comment:
I do not think this is a multiplier.

```suggestion
	 * @param light           packed lightmap coordinates
```